### ISSUE TITLE
'constructor' key is ignored with strict(), passthrough()

### ISSUE THREAD

**Issue by astef-trilogy** [colinhacks/zod#871]
This code doesn't throw:
```typescript
const person = z
  .object({
    name: z.string(),
  })
  .strict();

person.parse({
  name: "bob dylan",
  constructor: 61,
});
```

With other names, it throws. It is expected to throw with any property name.

It is also not being passed through (`.passthrough()`), so it's hard to prove that the library code actually receives such a property. But I think it does, since this is working:

```typescript
console.log(
  JSON.stringify({
    name: "bob dylan",
    constructor: 61,
  })
);

// outputs {"name":"bob dylan","constructor":61}
```

**Comment by JacobWeisenburger** [comment]
I was able to confirm this issue.

```ts
const person = z.object( { name: z.string() } ).strict()

// should throw, but doesn't
person.parse( { name: "bob dylan", constructor: 61 } )

// throws "Unrecognized key(s) in object: 'somethingElse'"
person.parse( { name: "bob dylan", somethingElse: 61 } )
```

**Comment by colinhacks** [comment]
Fixed in `zod@3.13`
